### ISSUE TITLE
GraphQL schema changes results in error: 'Attempting to mutate more than 1 global secondary index at the same time...'

### ISSUE THREAD

**Issue by duncangroenewald** [aws-amplify/amplify-cli#5558]
**Describe the bug**
When making schema changes to graphql I keep getting the following error:

```
Attempting to mutate more than 1 global secondary index at the same time on the VideoTable table in the Video stack. 
An error occurred during the push operation: Attempting to mutate more than 1 global secondary index at the same time on the VideoTable table in the Video stack. 
```
When that happens I have to basically blow away the entire schema by pushing a single table and once that is done replace the schema with the modified original and push all over again.

Is this something broken or am I doing something wrong?  Is there any workaround for this and this is a very time-consuming activity

**Amplify CLI Version**
4.29.6

**To Reproduce**
Modify your existing schema and change some of the @key

**Expected behavior**
I would expect the amplify-cli to be able to deal with this - and if necessary replace the dynamoDB table in order to build new partition keys.  

**Desktop (please complete the following information):**
 - OS: macOS
 - Node Version. n/a

**Additional context**
n/a

**Comment by yuth** [comment]
@duncangroenewald This is an known limitation of Cloudformation. This happens when either you're chainging the name of the `@key` or adding and removing `@key` directive on the same table. 

We are actively working on handling this automatically but it will take some time for us to get this fix in.

In the mean time the suggested workaround is to do phased deployment where the you can create an new index first  by adding an new `@key` and deploy it and then delete the old index by removing the `@key` directive which is no longer needed.

cc @SwaySway

**Comment by paddlefish** [comment]
The workaround is not practical if you use multiple environments, e.g. staging and production.

**Comment by duncangroenewald** [comment]
@yuth - thanks no problem for us right now - we just deploy an empty schema and then redeploy the full one - still learning the whole AppSync thing so having to refactor @connections etc since we have redundant IDs all over the place.  Some good documentation and examples would go a long way, there seem to be too many examples all doing things slightly differently.

Still impressive what can be done in a few days with Amplify, no need for DevOps anymore !

Now if everything was just typesafe wouldn't that be nice...

**Comment by AlessandroAnnini** [comment]
@yuth a proposal!
meanwhile you try to fix this problem would be nice to have a way to copy all the data from a connection to another before losing everything:

if I have
```
type Message @model {
  id: ID!
  author: User @connection(name: "UserMessage")
}
```

I could push for example
```
type Message @model {
  id: ID!
  author: User @connection(name: "UserMessage")
  editor: User @connection(name: "EditorMessage", copyFrom: "author")
}
```

so I won't loose all the relations, if the author field is already gone the `copyFrom` option should be ignored
then I will push
```
type Message @model {
  id: ID!
  editor: User @connection(name: "EditorMessage")
}
```

and the job is done without any blood spills :)

**Comment by SalahAdDin** [comment]
@AlessandroAnnini or at least a warning tutorial advicing how we must update/change schemas.

**Comment by cjihrig** [comment]
I'm going to close this out now that we have iterative deployments for this use case.
